mllib/src/test/scala/org/apache/spark/ml/classification/GBTClassifierSuite.scala

```diff
@@ -100,6 +101,40 @@ class GBTClassifierSuite extends SparkFunSuite with MLlibTestSparkContext {
     Utils.deleteRecursively(tempDir)
   }
 
+  test("GradientBoostedTrees should support all NumericType labels") {
+    val dfWithIntLabels = TreeTests.setMetadata(sqlContext.createDataFrame(Seq(
```

Comment:
I'm wondering if there is some way to reuse the code here since we need to implement roughly the same test (train with all different label column types, check that they are the same as training with double type) for each different predictor. Maybe adding something in a helper function under `MLTestingUtils`?